### PR TITLE
Treasury can pay out assets from the pallet-assets

### PR DESCRIPTION
**Pull request by clangenb**
Give the treasury the capability to pay out any asset that lives within the pallet-assets additionally to the native currency. The extrinsic used for that is the generic `spend` extrinsic, whose implementation is a bit different from the `spend_local` extrinsic that is solely used for payout out the native currency. See the documentation below:

```rust
//! ## Low Level / Implementation Details
//!
//! Spends can be initiated using either the `spend_local` or `spend` dispatchable. The
//! `spend_local` dispatchable enables the creation of spends using the native currency of the
//! chain, utilizing the funds stored in the pot. These spends are automatically paid out every
//! [`pallet::Config::SpendPeriod`]. On the other hand, the `spend` dispatchable allows spending of
//! any asset kind managed by the treasury, with payment facilitated by a designated
//! [`pallet::Config::Paymaster`]. To claim these spends, the `payout` dispatchable should be called
//! within some temporal bounds, starting from the moment they become valid and within one
//! [`pallet::Config::PayoutPeriod`].
```

tl;dr; the approved `spend`s need are not automatically paid out and need to be claimed within a claim period. 

Tested:

1. Propose a spend with the non-native currency by creating a preimage with the following call:
image

2. Claim the pending payout and if the treasury has enough money, the claim should succeed, and we see the following events:
image

Note: There is a conversion error in the UI in 2. because it assumes the unit AJUN, which means the number 1 maps 1'000'000 USDt/C because the AJUN has 6 more decimals than USDt/C.

Note2: I also tested that `spend_local` still works.

### CODE REVIEW

File: runtime/ajuna/src/benchmark_helpers.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	/// the `Beneficiary` for [`AccountId`].
+	///
+	/// This is slightly different from Polkadot's implementation, which uses a
+	/// [`VersionedLocatableAsset`] for the AssetId, as the asset lives on another chain.
+	pub struct TreasuryArguments<Parents = ConstU8<0>, ParaId = ConstU32<0>>(
+		PhantomData<(Parents, ParaId)>,
+	);
+	impl<Parents: Get<u8>, ParaId: Get<u32>> TreasuryArgumentsFactory<AssetKind, AccountId>
+		for TreasuryArguments<Parents, ParaId>
+	{
```
Comment (clangenb) [inline runtime/ajuna/src/benchmark_helpers.rs]: I just noticed that I can simplify this. We don't need the generic params like they do on polkadot.

**Review by DidacSF — APPROVED** [review]
